### ISSUE TITLE
Compilation errors in generated POJO code when generating <immutableInterfaces/> on a table accepting UDTs

### ISSUE THREAD

**Issue by superxiao** [jOOQ/jOOQ#17634]
### Expected behavior

Generated code can be compiled.

### Actual behavior

Generated code has compile error.

### Steps to reproduce the problem

I use the following sql migration:

```sql
CREATE TYPE task_metadata AS
(
  assigned_to TEXT,
  tags        TEXT[]
);

CREATE TABLE task
(
  id       SERIAL PRIMARY KEY,
  metadata task_metadata DEFAULT ROW (NULL, ARRAY []::TEXT[])::task_metadata
);
```

Use the following code gen config:
```
Generate().withFluentSetters(true).withImmutableInterfaces(true)
```

There is a compile error:

```
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler output below.
 .../src/main/jooq/org/jooq/generated/tables/pojos/Task.java:79: error: method does not override or implement a method from a supertype
      @Override
      ^
  .../src/main/jooq/org/jooq/generated/tables/records/TaskRecord.java:78: error: incompatible types: ITaskMetadata cannot be converted to TaskMetadataRecord
          setMetadata(from.getMetadata());
                                      ^
  .../src/main/jooq/org/jooq/generated/tables/pojos/Task.java:36: error: incompatible types: ITaskMetadata cannot be converted to TaskMetadata
          this.metadata = value.getMetadata();
                                           ^
  .../src/main/jooq/org/jooq/generated/tables/pojos/Task.java:84: error: cannot find symbol
              this.metadata = metadata.into(new TaskMetadata());
                                      ^
    symbol:   method into(TaskMetadata)
    location: variable metadata of type ITaskMetadata
```

### jOOQ Version

jOOQ opensource 3.19.15

### Database product and version

PostgreSQL 17

### Java Version

23

### JDBC / R2DBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql 42.7.4

**Comment by lukaseder** [comment]
Thanks for your report. I'll have a look at this, soon

**Comment by lukaseder** [comment]
I can reproduce the problem. Will look into this right away.

**Comment by lukaseder** [comment]
Fixed in versions:

- 3.20.0
- 3.19.16 (https://github.com/jOOQ/jOOQ/issues/17657)
- 3.18.23 (https://github.com/jOOQ/jOOQ/issues/17658)
- 3.17.32 (https://github.com/jOOQ/jOOQ/issues/17659)

**Comment by lukaseder** [comment]
(I'll confirm again if the backports really worked, once integration tests have passed. There have been numerous issues related to `<interfaces/>` in recent months, not all of which have been backported)

**Comment by lukaseder** [comment]
As I feared: The backport doesn't fix all issues that were reported in #17634 as some issues have been fixed via many other improvements that are available in 3.20 only.

The only workaround I can see right now is to avoid using `<interfaces/>`
